File: 01_projects/code/python-playwright/cbaps-dawms/robust/dawms/pages/reviewer_assignment_page.py
from playwright.sync_api import Page
from loguru import logger
from typing import List

class ReviewerAssignmentPage:
    def __init__(self, page: Page, pw):
        self.page = page
        self.pw = pw
        self.role_dropdown = page.locator("#reviewerRole")
        self.name_input = page.locator("#reviewerName")
        self.assign_btn = page.locator("button:has-text('Assign')")
        self.continue_btn = page.locator("button:has-text('Route to Signature')")
        self.pw.wait_for_selector("#reviewerRole")
    
    def assign_reviewer(self, data):
        logger.info(f"👤 Assigning: {data.name}")
        self.pw.select_dropdown(self.role_dropdown, data.role)
        self.pw.enter_text(self.name_input, data.name)
        self.pw.click_element(self.assign_btn)
        return self
    
    def assign_multiple_reviewers(self, reviewers: List):
        for reviewer in reviewers:
            self.assign_reviewer(reviewer)
        return self
    
    def route_to_signature(self):
        from .signature_routing_page import SignatureRoutingPage
        self.pw.click_element(self.continue_btn)
        return SignatureRoutingPage(self.page, self.pw)

File: 01_projects/code/java-playwright/cbaps-dawms/robust/src/main/java/com/playwright/dawms/pages/ReviewerAssignmentPage.java
package com.playwright.dawms.pages;
import com.microsoft.playwright.*;
import com.playwright.cbaps.library.EnhancedPlaywrightManager;
import com.playwright.dawms.models.ReviewerData;
import org.slf4j.*;
import java.util.List;

public class ReviewerAssignmentPage {
    private Page page;
    private EnhancedPlaywrightManager pwm;
    private Locator reviewerRoleDropdown, reviewerNameInput, assignReviewerButton, continueButton;
    
    public ReviewerAssignmentPage(Page page, EnhancedPlaywrightManager pwm) {
        this.page = page; this.pwm = pwm;
        this.reviewerRoleDropdown = page.locator("#reviewerRole");
        this.reviewerNameInput = page.locator("#reviewerName");
        this.assignReviewerButton = page.locator("button:has-text('Assign')");
        this.continueButton = page.locator("button:has-text('Route to Signature')");
        pwm.waitUntilElementVisible("#reviewerRole");
    }
    
    public ReviewerAssignmentPage assignReviewer(ReviewerData data) {
        pwm.selectDropdown(reviewerRoleDropdown, data.getRole());
        pwm.enterText(reviewerNameInput, data.getName());
        pwm.clickElement(assignReviewerButton);
        return this;
    }
    
    public void assignMultipleReviewers(List<ReviewerData> reviewers) {
        for (ReviewerData r : reviewers) assignReviewer(r);
    }
    
    public SignatureRoutingPage routeToSignatureStep() {
        pwm.clickElement(continueButton);
        return new SignatureRoutingPage(page, pwm);
    }
}
